### PR TITLE
feat: generate types for the AWS SDK

### PR DESCRIPTION
**Pull request by sam-goodwin**
In support of: https://github.com/functionless/functionless/pull/414

Generates an interface `SDK` that contains all of the AWS APIs in the form we want for the `$AWS.SDK`.

### CODE REVIEW

File: src/sdk-gen-aws.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import * as AWS from "aws-sdk";
+
+//
+export const aws = AWS;
+
+AWS.Wisdom;
```
Comment (sam-goodwin) [inline src/sdk-gen-aws.ts:6]: oops, typo

File: src/sdk-gen-aws.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import * as AWS from "aws-sdk";
+
+//
+export const aws = AWS;
```
Comment (sam-goodwin) [inline src/sdk-gen-aws.ts:4]: I grab the type of this Identifier in the gen script and use it to parse out all the types.

File: src/sdk-gen.ts [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+                    console.log(fqn);
+                    if (fqn === "__type") {
+                      // empty object {}
+                    } else if (
+                      checker.typeToString(responsePayloadType) === "any"
+                    ) {
+                      // ignore?
+                    } else if (fqn.indexOf(".") !== -1) {
+                      // <service-name>.<type-name>
+                      fqn;
```
Comment (sam-goodwin) [inline src/sdk-gen.ts:119]: at this point in the code we have the API method name, sevice name and request/response types. Last step is to simply generate the corresponding interfaces.

**Comment by sam-goodwin** [comment]
Build failing ... running out of memory?

```
<--- JS stacktrace --->

FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
 1: 0xb0a860 node::Abort() [node]
 2: 0xa1c193 node::FatalError(char const*, char const*) [node]
 3: 0xcf9a6e v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [node]
 4: 0xcf9de7 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [node]
 5: 0xeb1685  [node]
 6: 0xeb2166  [node]
 7: 0xec068e  [node]
 8: 0xec10d0 v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [node]
 9: 0xec404e v8::internal::Heap::AllocateRawWithRetryOrFailSlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [node]
10: 0xe8558a v8::internal::Factory::NewFillerObject(int, bool, v8::internal::AllocationType, v8::internal::AllocationOrigin) [node]
11: 0x11fe2d6 v8::internal::Runtime_AllocateInYoungGeneration(int, unsigned long*, v8::internal::Isolate*) [node]
12: 0x15f2d[39](https://github.com/functionless/functionless/runs/7941866950?check_suite_focus=true#step:4:40)  [node]
Aborted (core dumped)
```

**Review by thantos — APPROVED** [review]

**Pull request merged**